inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp

```diff
@@ -184,6 +248,105 @@ struct jit_uni_mvn_mean_variance_kernel_f32 : public jit_uni_mvn_mean_variance_k
     Xbyak::Xmm xmm_aux1 = Xbyak::Xmm(3);
     Xbyak::Xmm xmm_aux2 = Xbyak::Xmm(4);
     Xbyak::Xmm xmm_aux3 = Xbyak::Xmm(5);
+    Vmm vmm_zero = Vmm(6);
+
+    Xbyak::Opmask k_mask = Xbyak::Opmask(7);
+
+    std::unique_ptr<jit_load_emitter> load_emitter = nullptr;
+
+    std::vector<size_t> load_pool_gpr_idxs;
+
+    inline void worker_full_size(load_emitter_context &load_ctx) {
+        load_emitter->emit({static_cast<size_t>(reg_src.getIdx())}, {static_cast<size_t>(vmm_val.getIdx())},
+                            {}, {load_pool_gpr_idxs}, emitter_in_out_map::gpr_to_vec, static_cast<emitter_context*>(&load_ctx));
```

Comment:
To be honest from code readability perspective it doesn't look very well. Moreover we create ctx separatelly. I would have expected smt like:
`
 load_emitter->emit({reg_src.getIdx()}, {vmm_val.getIdx()},
                            {}, {load_pool_gpr_idxs}, {src_prc, dst_prc, step});
`

Now we can see which exact parameters (precisions and tiles) are using by particular emit call. Key things:
1. We need to remove casts. For indexes seems like we need to change base ctor to accept `int` instead of `size_t`.  For  ctx seems like we should cast it to derived class inside emitter.
2. We need to be able to construct ctx inplace and pass it as rvalue. So seems like we need ctor with const reference and initializer list here. 
@maxnick  what do you think about the proposal? Could you please advice how we can achieve such level os usability via emitter API?

Reply:
In the current paradigm it is hard to force an user to instantiate the context object in the call inplace. Since we pass the context to the emit call as a base type we have only two possibilities here: pass the context as a pointer or as a reference. Neither of them can force the user to construct the context only in the call itself, but both two approaches allow to do this. But using a pointer gives little bit more flexibility, we can pass nullptr in special cases or by default. 
If we use reference to pass the context, emit call may look like as the following:
`load_emitter->emit({reg_src.getIdx()}, {vmm_val.getIdx()}, {}, {load_pool_gpr_idxs}, load_emitter_context{src_prc, dst_prc, step})`,
If we use pointer:
`load_emitter->emit({reg_src.getIdx()}, {vmm_val.getIdx()}, {}, {load_pool_gpr_idxs}, &(load_emitter_context{src_prc, dst_prc, step}))`,
In both cases we have to explicitly specify the context type since in the former case the emit call expect the base class reference and passing just initialization list will force the compiler to instantiate the base context object from the initialization list that in turn will lead to an error.

Reply:
currently done by [e7e9199](https://github.com/openvinotoolkit/openvino/pull/3567/commits/e7e919973687383d0758904aef7f00079aa911ba)
1. the index type is changed to int for all base and derived emitters and removed cast.
2. besides @maxnick 's comments -- explicit context type is needed. If we need create context and call emit in one sentence, we must pass const reference into emit(). Because local const reference will prolong the lifetime of temporary object until the end of containing scope, while pointer can not and will have compiler error of "taking address of temporary".  Now it looks like [load_emitter->emit({reg_src.getIdx()}, {vmm_val.getIdx()}, {}, {load_pool_gpr_idxs}, load_emitter_context{jcp_.src_prc, dst_prc, step});](https://github.com/openvinotoolkit/openvino/pull/3567/commits/e7e919973687383d0758904aef7f00079aa911ba#diff-3593bcddbeb921a60602fad8865d89c2b8f74901bc4a28114c7bb976f625f8f7R236)

Reply:
Sorry, I was wrong indeed. My bad. Haven't used raw pointers for a long time. 
But the idea of using pointers still remains. We just could use shared pointers with a slightly uglier syntax (std::make_shared call) and little runtime overhead, but with the special null state, which might be useful, since in that case we can unambiguously determine that it is a default call.
In case we use a constant reference with an instance of the base class as the default, it is difficult to distinguish a default call from a call with the context being passed and in that case it is better to have a separate emit interface method without context parameter.

Reply:
Agree that nullptr can determine if default call. Now shared_ptr is used. Thank you. Take a look again please in bound of this commit [d2d0e67](https://github.com/openvinotoolkit/openvino/pull/3567/commits/d2d0e671d7f854028494305fb6ab1b5e60fcb5ac)
API looks like: 
[    virtual void emit(const std::vector<int> &in_idxs, const std::vector<int> &out_idxs,
                      const std::vector<int> &pool_vec_idxs = {}, const std::vector<int> &pool_gpr_idxs = {},
                      const std::shared_ptr<emitter_context> &emit_context = nullptr);](https://github.com/openvinotoolkit/openvino/pull/3567/commits/d2d0e671d7f854028494305fb6ab1b5e60fcb5ac#diff-759fdf3e56d61e5d6c489b7c8bb36a952c2f37db04339fce36b47e6427fcf618R33)
How to use:
        [load_emitter->emit({reg_src.getIdx()}, {vmm_val.getIdx()},
                            {}, {load_pool_gpr_idxs},
                            std::make_shared<load_emitter_context>(jcp_.src_prc, dst_prc, step));](https://github.com/openvinotoolkit/openvino/pull/3567/commits/d2d0e671d7f854028494305fb6ab1b5e60fcb5ac#diff-3593bcddbeb921a60602fad8865d89c2b8f74901bc4a28114c7bb976f625f8f7R236)

Reply:
Yep, looks good. Passing shared_ptr as a const reference always works but the function can change the state of what it points to. So if the semantics of the emit call does not allow changing the emitter_context state, it is better to use shared pointer to a const type `virtual void emit( ... , const std::shared_ptr<const emitter_context> &emit_context = nullptr);` The usage of the emit call will remain the same since the constness modifier is added implicitly. 
The semantics of the context usage is an open question though.

Reply:
For now I cannot imagine situations when we will need to change the ctx inside emitter. Lets keep current version which is based on shared_ptr. @chenhu-wang @maxnick big thanks for constructive discussion.

Reply:
const reference of shared_ptr to const emitter_context is used now by [b1a1cfb](https://github.com/openvinotoolkit/openvino/pull/3567/commits/b1a1cfbd2a1e85dcf05d563b3c6f4b525b70dd54)